datalad/distributed/create_sibling_gitlab.py

```diff
@@ -0,0 +1,598 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""High-level interface for creating a publication target on a GitLab instance
+"""
+
+__docformat__ = 'restructuredtext'
+
+
+import logging
+
+from ..interface.base import (
+    build_doc,
+    Interface,
+)
+from ..interface.common_opts import (
+    recursion_flag,
+    recursion_limit,
+    publish_depends,
+)
+from ..interface.utils import eval_results
+from ..support.param import Parameter
+from ..support.constraints import (
+    EnsureChoice,
+    EnsureNone,
+    EnsureStr,
+)
+from ..utils import (
+    assure_list,
+)
+from ..distribution.dataset import (
+    datasetmethod,
+    EnsureDataset,
+    require_dataset,
+    rev_resolve_path,
+)
+from ..dochelpers import exc_str
+
+# bound methods
+from ..distribution.siblings import Siblings
+from ..local.subdatasets import Subdatasets
+
+
+lgr = logging.getLogger('datalad.distributed.create_sibling_gitlab')
+
+known_layout_labels = ('hierarchy', 'collection', 'flat')
+known_access_labels = ('http', 'ssh', 'ssh+http')
+
+
+@build_doc
+class CreateSiblingGitlab(Interface):
+    """Create dataset sibling at a GitLab site
+
+    A Git repository can be created at any location/path a given user has
+    appropriate permissions for. API access and authentication are implemented
+    via python-gitlab, and all its features are supported. A particular GitLab
+    site must be configured in a named section of a python-gitlab.cfg file
+    (see https://python-gitlab.readthedocs.io/en/stable/cli.html#configuration
+    for details), such as::
+
+      [mygit]
+      url = https://git.example.com
+      api_version = 4
+      private_token = abcdefghijklmnopqrst
+
+    Subsequently, this site is identified by its name ('mygit' in the example
+    above).
+
+    (Recursive) sibling creation for all, or a selected subset of subdatasets
+    is supported. Three different project layouts for nested datasets are
+    supported (see --layout):
+
+    "hierarchy"
+      Each dataset is placed into its own group, and the actual GitLab
+      project for a dataset is put in a project named "_repo_" inside
+      this group. Using this layout, arbitrarily deep hierarchies of
+      nested datasets can be represented, while the hierarchical structure
+      is reflected in the project path. This is the default layout, if
+      no project path is specified.
```

Comment:
Based on the code

```python
    if layout is None:
        # figure out the layout of projects on the site
        # use the reference dataset as default, and fall back
        # on 'hierarchy' as the most generic method of representing
        # the filesystem in a group/subgroup structure
        layout_var = 'datalad.gitlab-{}-layout'.format(site)
        layout = ds.config.get(
            layout_var, refds.config.get(
                layout_var, 'hierarchy'))
```

it seems always be the default even if a project is specified.